packages/flutter/lib/src/services/asset_manifest.dart

```diff
@@ -104,4 +121,11 @@ class AssetVariant {
   /// The asset's key, which is the path to the asset specified in the pubspec.yaml
   /// file at build time.
   final String key;
+
+  /// Whether or not this is a main asset. In other words, this is true if
+  /// this asset is not a variant of another.
+  ///
+  /// [Asset variants](https://docs.flutter.dev/development/ui/assets-and-images#asset-variants)
+  /// for more about asset variants.
+  final bool isMainAsset;
```

Comment:
In general, we try to give properties fairly simple names and rely on that name + the type name to give full context. For example, instead of naming a property on Semantics `bool isSemanticsButton` we'd just call it `bool button`. In this case, I think the asset part is implied by the type being called `AssetMetadata` so you could shorten this to `main` and add more context via the doc comment.